data/helpers.d/hardware

```diff
@@ -53,20 +55,46 @@ ynh_check_ram () {
 			# Use only the amount of free swap
 			ram=$total_swap
 		fi
+    else
+        echo "Uhoh, you should choose --free or --total when using ynh_get_ram" >&2
+        ram=0
 	fi
 
-	if [ -n "$required" ]
-	then
-		# Return 1 if the amount of ram isn't enough.
-		if [ $ram -lt $required ]
-		then
-			return 1
-		else
-			return 0
-		fi
+    echo $ram
+}
 
-	# If no RAM is required, return the amount of available ram.
-	else
-		echo $ram
-	fi
+# Return 0 or 1 depending if the system has a given amount of RAM+swap free or total
+#
+# usage: ynh_require_ram [--amount=RAM required in Mb] [--free|--total] [--ignore_swap|--only_swap]
+# | arg: -a, --amount - The amount to require, in Mb
+# | arg: -f, --free  - Count free RAM+swap
+# | arg: -t, --total - Count total RAM+swap
+# | arg: -s, --ignore_swap - Ignore swap, consider only real RAM
+# | arg: -o, --only_swap - Ignore real RAM, consider only swap
+ynh_require_ram () {
+    # Declare an array to define the options of this helper.
+	declare -Ar args_array=( [a]=amount= [f]=free [t]=total [s]=ignore_swap [o]=only_swap )
+    local amount
+    local free
+    local total
+	local ignore_swap
+	local only_swap
+	# Manage arguments with getopts
+	ynh_handle_getopts_args "$@"
+	amount=${amount:-0}
+    # Dunno if that's the right way to do, but that's some black magic to be able to
+    # forward the bool args to ynh_get_ram easily?
+    free=${free:+--free}
```

Comment:
It works indeed. Same as :- except that it replace the content only if the variable **is** set.